---
title: huangChickenGutMetagenome2018
author: ''
date: '2023-01-12'
slug: ['huangChickenGutMetagenome2018']
categories: []
tags:
  - one tag
  - another tag
meta_img: images/image.png
description: Description for the page
---
# The chicken gut metagenome and the modulatory effects of plant-derived benzylisoquinoline alkaloids
#### (2018) - Peng Huang, Yan Zhang, Kangpeng Xiao, Fan Jiang, Hengchao Wang, Dazhi Tang, Dan Liu, Bo Liu, Yisong Liu, Xi He, Hua Liu, Xiubin Liu, Zhixing Qing, Conghui Liu, Jialu Huang, Yuwei Ren, Long Yun, Lijuan Yin, Qian Lin, Cheng Zeng, Xiaogang Su, Jingyang Yuan, Li Lin, Nanxi Hu, Hualiang Cao, Sanwen Huang, Yuming Guo, Wei Fan, Jianguo Zeng
- **Link**:: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6260706/
- **DOI**:: 10.1186/s40168-018-0590-5
- **Links**:: 
- **Tags**:: #paper
- **Cite Key**:: [@huangChickenGutMetagenome2018] 

### Abstract
Background Sub-therapeutic antibiotics are widely used as growth promoters in the poultry industry; however, the resulting antibiotic resistance threatens public health. A plant-derived growth promoter, Macleaya cordata extract (MCE), with effective ingredients of benzylisoquinoline alkaloids, is a potential alternative to antibiotic growth promoters. Altered intestinal microbiota play important roles in growth promotion, but the underlying mechanism remains unknown. Results We generated 1.64 terabases of metagenomic data from 495 chicken intestinal digesta samples and constructed a comprehensive chicken gut microbial gene catalog (9.04 million genes), which is also the first gene catalog of an animal’s gut microbiome that covers all intestinal compartments. Then, we identified the distinctive characteristics and temporal changes in the foregut and hindgut microbiota. Next, we assessed the impact of MCE on chickens and gut microbiota. Chickens fed with MCE had improved growth performance, and major microbial changes were confined to the foregut, with the predominant role of Lactobacillus being enhanced, and the amino acids, vitamins, and secondary bile acids biosynthesis pathways being upregulated, but lacked the accumulation of antibiotic-resistance genes. In comparison, treatment with chlortetracycline similarly enriched some biosynthesis pathways of nutrients in the foregut microbiota, but elicited an increase in antibiotic-producing bacteria and antibiotic-resistance genes. Conclusion The reference gene catalog of the chicken gut microbiome is an important supplement to animal gut metagenomes. Metagenomic analysis provides insights into the growth-promoting mechanism of MCE, and underscored the importance of utilizing safe and effective growth promoters. Electronic supplementary material The online version of this article (10.1186/s40168-018-0590-5) contains supplementary material, which is available to authorized users.

### Notes
<h1>Annotations
 (1/12/2023, 3:26:31 PM)</h1> 

“Probiotics are living microorganisms that confer benefits to the host, examples being Lactobacillus and Bifidobacterium” (Huang et al., 2018, p. 2) 

“The phytobiotic Macleaya cordata extract (MCE) has been widely used, for decades, in feed livestock in many countries [9]. The effective chemical composition of MCE includes sanguinarine and chelerythrine, both belonging to a group of benzylisoquinoline alkaloids, which have antimicrobial and anti-inflammatory properties [10–12].” (Huang et al., 2018, p. 2) 

“In this study, we constructed the first comprehensive chicken gut microbial gene catalog, to better understand the related microbiota” (Huang et al., 2018, p. 2) 

“We collected 495 digesta samples from the five intestinal compartments (duodenum, jejunum, ileum, cecum, and colorectum) of chickens raised in 7 different farms in China” (Huang et al., 2018, p. 2) 

“Based on the assembled contigs with an N50 contig length of 1.95 kb, we identified 9.04 million non-redundant genes, with an average open reading frame (ORF) length of 697 bp.” (Huang et al., 2018, p. 2) 

“By comparing the pairwise overlap, at the gene sequence level, we determined that over 80% of genes are unique to each species, and only a very small percentage (~ 0.5%) of genes are shared by chickens, humans, and pigs.” (Huang et al., 2018, p. 2) 

“Using CAMRA3 for taxonomic assignment, 80.99% of the genes in the chicken gut catalog were taxonomically classified at the superkingdom level, among which bacteria account for 98.95% of the classified genes, with the remaining 1% being from archaea and eukaryotes. More than 88% of the bacterial genes are from the top four phyla, including Firmicutes, Actinobacteria, Proteobacteria, and Bacteroidetes” (Huang et al., 2018, p. 2) 

“t lower taxonomic levels, 25.97% and 2.29% of the genes in this catalog were annotated to genus and species, respectively.” (Huang et al., 2018, p. 2) 

“Note that the short-chain fatty acid (SCFA)-producing genera, such as Bacteroides, Blautia, Ruminococcus, and Faecalibacterium, are among the major genera in both human” (Huang et al., 2018, p. 2) 

“Notably, the genes for glycan biosynthesis and metabolism were more abundant in human and pig guts, whereas” (Huang et al., 2018, p. 4) 

“whereas the genes for membrane transport, an essential mechanism for the uptake of substrates, such as sugars, lipids, peptides, and ions, were more abundant in the chicken gut microbiota (Fig. 1f ).” (Huang et al., 2018, p. 4) 

“The enriched genes for membrane transport likely reflect the availability of more nutrient substrates, in the chicken intestine, for direct microbial utilization. The more abundant genes for metabolism of xenobiotics, terpenoids, and polyketides, in the chicken gut, are relevant to the abundant bacterial phylum Actinobacteria, which decomposes organic matter and produces various natural drugs, enzymes, and bioactive metabolites” (Huang et al., 2018, p. 4) 

“Distinguished by the difference in morphology and function, the chicken intestinal tract can be divided into the foregut and hindgut.” (Huang et al., 2018, p. 4) 

“The foregut contains the duodenum, jejunum, and ileum compartments. The duodenum has the major function in feed digestion by using digestive enzymes and bile from the pancreas and liver, and the released nutrients, such as amino acids, fatty acids, sugars, and peptides, are mainly absorbed in the jejunum and ileum.” (Huang et al., 2018, p. 4) 

“The hindgut contains the cecum and colorectum. Substantial microbial fermentation occurs in the cecum, which provides nutrients, detoxifies some harmful substances, and also helps to prevent pathogen colonization [20,21].” (Huang et al., 2018, p. 4) 

“Previous chicken gut studies focused more on cecal or fecal microbiota [23–25], and some microbial functions such as polysaccharide utilization, SCFAs production, and hydrogen consumption in the cecum have been studied” (Huang et al., 2018, p. 4) 

“he microbiota in the foregut where nutrient absorption primarily occurs were mostly studied by 16S rRNA gene sequencing with limited sample size” (Huang et al., 2018, p. 4) 

“In this part of our study, metagenomic data of 285 samples from all five intestinal compartments in chickens older than 40 days were analyzed (Fig. 1a and Additional file 1: Table S1).” (Huang et al., 2018, p. 4) 

“Our results indicated that microbial diversities in the foregut compartments were approximately the same, with only a slight increase from duodenum, jejunum, to ileum” (Huang et al., 2018, p. 6) 

“A similar situation was observed between the hindgut cecum and colorectum; however, the diversities were clearly higher in the hindgut than in the foregut compartment (Fig. 2a)” (Huang et al., 2018, p. 6) 

“We next calculated the number of common genes presented in 50% of the samples from each intestinal compartment, and the results showed that the common genes accounted for only about 2% of all genes in each foregut compartment (duodenum, jejunum, and ileum), but the number increased to 18% and 13% in the cecum and colorectum, respectively (Additional file 12: Table S4)” (Huang et al., 2018, p. 6) 

“Taken together, these results indicated an overall similarity within foregut and hindgut compartments, with a larger difference between them, which is consistent with previous studies” (Huang et al., 2018, p. 6) 

“Lactobacillus is the predominant genus in the foregut, but not in the hindgut (Additional file 4: Table S3); Lactobacillus provides nutrients to the host and defends against opportunistic pathogens” (Huang et al., 2018, p. 6) 

“n the foregut, Lactobacillus competitively inhibits a cluster of bacteria, with negative correlations with all these genera” (Huang et al., 2018, p. 6) 

“Additionally, some SCFA producers, such as Clostridium, Butyricicoccus,andFaecalibacterium, showed positive correlations with one another and form a relatively independent and stable cluster” (Huang et al., 2018, p. 6) 

“These results revealed more diverse and complex microbial communities in the hindgut than in the foregut.” (Huang et al., 2018, p. 6) 

“the microbiota in the foregut was enriched in genetic information processing for transcription, translation, and replication, as well as the metabolic functions of nucleotides and lipids, whereas microbes in the hindgut were enriched for the metabolic functions of amino acids, energy metabolism, and secondary metabolite biosynthesis.” (Huang et al., 2018, p. 6) 

“Lactobacillus has a relatively small genome (2 Mb) compared to other bacteria, and encodes higher proportion of genes for basic functions such as transcription, translation, and replication, but lower proportion of genes for many diverse metabolic functions. Therefore, the predominance of Lactobacillus in the foregut has largely contributed to the functional differences between the foregut and hindgut microbiome.” (Huang et al., 2018, p. 6) 

“Our data showed that the gut microbiome of newly hatched chicks (day 1) were highly variable and obviously different from other samples (day 7, 14, 28, and 42) (Fig. 3), reflecting the shortterm exposure to the environment and the initiation of gut microbial communities.” (Huang et al., 2018, p. 8) 

“The findings showed that the early days are critical both for chicken development and establishment of the gut microbiota.” (Huang et al., 2018, p. 8) 

“In the MCE groups, a greater variety of pathways were enriched, including lipoic acid, vitamin B6, riboflavin, ubiquinone, and other terpenoid-quinone (including vitamin K1,K2, and E) pathways (Fig. 4d). In summary, vitamins, in addition to other microbial synthesized nutrients, were enhanced by MCE and consequently benefited the host.” (Huang et al., 2018, p. 11) 

“The reads that mapped to chicken, human, maize, soybean, wheat,” (Huang et al., 2018, p. 13) 

“and zebrafish genomes by BWA-MEM were filtered out” (Huang et al., 2018, p. 13) 

“Finally, short reads (length &lt; 75-bp) and unpaired reads were also excluded to form a clean reads data.” (Huang et al., 2018, p. 13) 

“For each sample, the clean reads were assembled by Megahit (v1.0.6) under pair-end mode respectively [73], then gene prediction was performed on contigs larger than 500-bp by Prodigal (v2.6.3) with parameter “-p meta” [74], and gene models with cds length less than 102-bp were filtered out.” (Huang et al., 2018, p. 13) 

“due to the memory limitation of our computer server, all the 495 samples were firstly divided into 5 study groups: distribution groups (135 samples), treatment groups for LY chickens with 56 days (75 samples), treatment groups for LY chickens with 1–42 days (105 samples), treatment groups for AA chickens with 42 days (75 samples), and treatment groups for AA chickens with 1–42 days (105 samples). Then, assembly and gene prediction were performed on these five study groups independently, using the same methods for each sample.” (Huang et al., 2018, p. 13) 

“A non-redundant gene catalog was constructed using the gene models predicted from each sample and each group by cd-hit-est (v4.6.6) [75] with parameter “-c 0.95 -n 10 -G 0 -aS 0.9,” which adopts a greedy incremental clustering algorithm and the criteria of identity &gt; 95% and overlap &gt; 90% of the shorter genes. By using the gene models predicted from each sample only, we obtained 6 M non-redundant genes; by adding the additional gene models predicted from each group, finally, we obtained a total of 9 M non-redundant genes.” (Huang et al., 2018, p. 13) 

“Taxonomic assignments of protein sequences were made on the basis of DIAMOND (v0.8.28.90 diamond blastp –evalue 10 –max-target-seqs 250) alignment against the NCBI-NR database by CARMA3” (Huang et al., 2018, p. 13) 

“3 (carma –classify-blast –type p –database p)” (Huang et al., 2018, p. 13) 

“The” (Huang et al., 2018, p. 14) 

“he significant functional differences between the chicken, human, and pig gut samples were determined by the Wilcoxon rank-sum test, adjusted by the Storey’s methods for multiple tests.” (Huang et al., 2018, p. 14) 

“To determine the taxonomic and functional differences between the foregut and hindgut microbial communities, the Wilcoxon rank-sum test (Storey’s methods for multiple tests adjustment) was applied. For the comparison of treatment groups, the Kruskal-Wallis test (Storey’s methods for multiple tests adjustment) was applied, followed by the post-hoc Wilcoxon rank-sum test” (Huang et al., 2018, p. 14)