---
title: 'Module 11'
date: 2023-07-24T12:44:58Z
draft: false
meta_img: "images/image.png"
tags:
  - "stat6315"
  - "microbiome-review"
  - "kegg ontology network"
  - "gg-catalog"
  - "Meet-With-Aggrey"
description: "Description for the page"
---

### Todos for Today

- Make up TAP forms for Dr. Aggrey to sign
- Read papers about microbiome analysis
- STAT 6315
  - Watch Module 11
  - Finish Module 11 Homework
- fully ingest Knight et al 2018
- Jackwood Blast
  - meet Ben and Brian TBD
  - try out a local blast search to see memory, cpu time limitation
- gg-catalog
  - Implications of the networks generated
  - How does KEGG work?
  - Show Aggrey the charts and ask him what he thinks
  - Find a Shotgun Analysis involved with chickens and some kind of phenotypic data.
  - Some kind of machine learning classifier to determine major players in a phenotypic outcome.

### STAT 6315

#### Module 11

11.1a tumors part 1

odds of sucess is p/(1-p)

odds > 1, probability > 0.5 and vice versa

odds ratio is for compatative purposes 

odds for one or more tumors for low dose is 1.468 times the odds of more more more tumor for the control 

odds for one or more tumors for the high dose is 2.108 times the odds of one or more turmors for the control.

odds ratio > 1 => p1 > p2

compare the odds ratios to 1

11.1b tumors part 2

logistic regression can model probability of outcome for binary outcome.

p/(1-p) is odds. 

natural log of the odds. 

log odds are linearly related between control and high

indicator variables, remember they are 0 or 1, only need 2 because there are three categories, so k-1 indicators

predicted log odds vs predicted probability

-1.815 - 0.382 (1) + 0.365 (0) = -2.197 | e^(-2.197)/ 1+e^(-2.197) = .10002

-1.815 - 0.382 (0) + 0.365 (0) = -1.815 | e^(-1.815) / 1+e^(-1.815) = .14003

-1.815 - 0.382 (0) + 0.365 (1) = -1.45 | e^(-1.45) / 1+e^(-1.45) = .190002

the predicted probabilities match up with what we calculated by hand, use JMP to fit model, find log odds and then find predicted probabilites from that. 

log odds are linearly related through the jmp coeffecients


11.1c interpretation

predictor is quantitative, response is categorical

exp(ln(odds)) = exp(b0+bi x)

odds = exp(b0 + b1 x)

exp(b0 + b1 (x+1))

for every 1 unit increase in x....

exp(b0 + b1 (x+1))/exp(b0+b1 x) = exp(b0) exp(b1 x) exp(b1)/ exp(b0) exp(b1 x) = exp(b1) 

for every one unit increase of x, the odds of sucess multiples by e to the b1

exp(b0 + b1 (x+2))/exp(b0+b1 x) = exp(b0) exp(b1 x) exp(2 b1)/ exp(b0) exp(b1 x) = exp(2 b1)


11.1d space shuttle challenger part 1

lhat = log odds of one o-ring failing = b0 + b1 (temp)

lhat = 15.043 - 0.232(temperature)


find temperature for at least 50% chance oring fails

p = 0.5, how do I get from p to log odds 

ln(p/1-p) = log odds = ln(0.5/1-0.5) = ln(1) = 0 

0 = 15.043 - 0.232 (temp) 

temp = -15.043/-.232 = 64.84

for a temperature of 64.84 there is a 50% chance of at least one o-ring failing. higher temp less than 50% chance, lower temp increased chance of o-ring failing. 


11.1e space shuttle challenger part 2

b1 = -0.232 => e^-0.232 = 0.793 for every 1 degree increase in temperature the odds that at least one o-ring fails multiplies by 0.793. 

odds are decreasing because odds ratio is less than 1 

instead of 1 unit increase, 28 unit increase: 81-53 = 28, e^(28*-0.232) = 0.0015; the odds of at least one o-ring failing multiplies by 0.0015. 


lhat = 15.043 - 0.232 (31) = 7.851 

e^(7.851)/1+e^(7.851) = 0.9996, very high chance that one o-ring will fail

11.2a diamonds

the predicted log odds is lhat = -0.916 + 1.204 (0) + 0.805 (0) + 1.239 (0) = -0.916

the predicted probabilty that the price is more than 3,000 = e^(-0.916)/1+e^(-0.916) = 0.286


e^(0.323)/ 1+ e^(0.323) = 0.58, for a diamond that has an excellent cut there is a 0.58 probability the price is greater than 3,000

the predicted log odds is lhat = -0.916 + 1.204 (0) + 0.805 (0) + 1.239 (1) = 0.323


odds ratio: 

odds for excellent = e^0.323 =  1.381
odds for ideal = e^-0.916 = 0.4

odds ratio = 1.381/0.4 = 3.45 the odds an excellent cut diamond has a price of more than 3,000 is 3.45 times the odds an ideal cut diamond has a price of more than 3,000 

11.2b pesticide

fit a model or equation: log odds = lhat = b0 + b1 (dosage)

ln(prop/1-prop) 

lhat = -3.209 + 0.951 (dosage)

what is ld 50

p = 0.5 log odds = ln(0.5/1-.5) = ln(1) = 0 

0 = -3.209 + 0.951 (dosage) = 3.374

about 50% of the insects will be killed when using a dosage of 3.374

the predicted log odds = -3.209 + 0.951 ( 2.5) = -0.8315 then the odds is e^(-0.8315) = 0.435, the probabiltiy an insect is killed with this dosage is 0.435 times the probability it is not killed. 

p/1-p = 0.435 => p = 0.453(1-p)  

3-1 = 2 
e^(2 * b1(0.951)) =  6.7 for a dosage of 3, the odds an insect is killed is 6.7 times the odds an insect is killed using a dosage of 1. 

11.2c urology

log odds = lhat = 1.735 - 1.344 (0) + 0.345 (1) + 0.645 (0) + 0.289 (0 \* 1) + 1.181 (0 \* 0)

1.735 + 0.345 = 2.08

the probabilty they are cured = e^(2.08) /1+e^(2.08) = 0.889

log odds = lhat = 1.735 - 1.344 (1) + 0.345 (0) + 0.645 (0) + 0.289 (0 \* 0) + 1.181 (0 \* 0)

1.735-1.344 = 0.391

e^(0.391)/1+e^(0.391) = 0.597 

11.2d myopia

lhat = -1.239 - 1.721 (0) - 0.039 (10) + 0.015 (2) = -1.599

p(the child is myopic) = e^(-1.599)/1+e^(-1.599) = 0.168

e^(-1.599) = 0.202

0.168/1-0.168 = 0.202

the probability that this child is myopic, is 0.202 time the probability they are not myopic

lhat = b0 + b1(P) + b2(s) + b3(c)

-1.239 - 1.721 (0) - 0.039 (S) + 0.015 (C) 
-1.239 - 1.721 (1) - 0.039 (S) + 0.015 (C) 

odds the child is myopic is e^(lhat) = e^b0 \* e^b1(p) \* e^b2(s) \* e^b3(c)

odds myopic if parents are myopic/odds myopic if parents are not myopic

e^b0 \* e^b1(0) \* e^b2(s) \* e^b3(c) / e^b0 \* e^b1(1) \* e^b2(s) \* e^b3(c) = e^b1(0) /  e^b1(1) = 1/e^(-1.721) = e^1.721 = 5.59

the odds a child is myopic if parents are myopic is 5.59 times the odds the child is myopic if parents are not myopic. for a specified number of sports and computer hours. Sports and computer hours must stay fixed.

#### Homework 11

I completed and submitted the homework 11. 


### Dr. Aggrey

I found my old notes on page (03-13-2023 Example Clean Read Mapping):

  [Meet Dr. Aggrey](https://lorentznotebook.netlify.app/blog/2023-03-13-example-clean-read-mapping/)

  I wanted to check in with Dr. Aggrey and see if I am on the right path. It seems that I am a little lost. We talked through the data that I have and he suggested I get an easy victory. First he wanted me to identify the genes from the Huang data that are associate with Nitrogen Metabolism. Then he wants me to establish a gene network from those genes. I am not sure what I should do with that gene network once it is established.

  Step 2 I was thinking of was to then take those genes of interest and try to find a FCR efficiency study that uses shotgun reads and then examine the abundance of my genes of interest and see if that is correlated with FCR.

I added a note to the word explanation stating that we had discussed me generating a gene network and the resulting keggs I have shared is my first attempt. Is that something like he expected, and what characteristics is he looking for? 

Okay much better meeting with Dr. Aggrey and I have the next 3 steps set up for analysis, the first thing to remember is "Microbial Genes Associated with Nitrogen Metabolism"

the steps are as follows:

1. Identify the major pathways of nitrogen metabolism
  - Identify the Kegg pathways
  - Identify the major genes (20-25) for each tissue in each pathway 

2. Generate Table of Relative abundance
  - does the abundance differ between gut segment

3. Functional analysis
  - go through the literature and see are these most abundant genes catabolism or anabolism oriented?

### Papers

Ingest Knight et al 2018

Imported and have some headers created for best practices and current struggles


### Todos for Tomorrow

- Make up TAP forms for Dr. Aggrey to sign Next 2 needed
- Read papers about microbiome analysis
- STAT 6315
  - Watch Module 12
  - Finish Module 12 Homework
  
- gg-catalog
  - Identify KEGG pathways that deal with Protein Metabolism/Nitrogen metabolism
  
- Jackwood Blast
  - meet Ben and Brian TBD
  - try out a local blast search to see memory, cpu time limitation

### Git Commits

#### Lab Notebook

```bash
73b34b3 - Benjamin Lorentz, Mon Jul 24 14:37:13 2023 -0400 : add updates post lunch
a7e2f6a - Benjamin Lorentz, Mon Jul 24 11:55:30 2023 -0400 : notes before lunch
94530a7 - Benjamin Lorentz, Mon Jul 24 08:48:38 2023 -0400 : added a page for monday
```

#### gg-catalog

```bash
8193f3c - Benjamin Lorentz, Mon Jul 24 16:25:38 2023 -0400 : add 04_huang_kegg_git_tables.rmd
```
