---
title: 'Primer Detection Benchmark'
date: 2023-05-26T13:48:22Z
draft: false
meta_img: "images/image.png"
tags:
  - "Ade"
  - "visualize-ampliseq"
  - "Mock"
  - "Module"
  - "primer-detect"
description: "Description for the page"
---

### Todos for Today

- Jackwood Blast
  - meet Ben and Brian TBD
  - try out a local blast search to see memory, cpu time limitation
- Ade
  - compare fastp and prinseq analyses
  - How does the other Ben's Analysis line up with mine/ampliseq?
    - filtering step for the abundance?
    - Filtering unknown taxa?
  - Run a proper analysis to send to Ade
- gg-catalog
  - Generate a gene network 
    - how do you do this?
      - possibly this: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-3371-7
  - query the KO list and cross ref to the abundance data
  - Find a Shotgun Analysis involved with chickens and some kind of phenotypic data.
  
### Ade

#### Rebuild Figaro for Quay.io

#### Fastp

cycle 4 rev: b24fee34c6208d1f815578670ffd2ff65ed75836
slurm job: 23019292

```bash
```

#### Setup new analysis with both cleaned datasets

Prinseq Analysis

cycle 4 rev: 5563d66487fa4bd89861d8f7a0704d8836c7c20f
visualize ampliseq srs rev: c6b07c621749576e0cd35813d7861d2a5573da36
slurm job: 

```bash
```

#### Ben's Other analysis

What steps did he take?

What steps do I take?



#### Longitudinal Analysis

See if it makes a difference in this specific analysis. 


### Questions to ask for new analysis

- What primers did you use for amplification?
- What region were you targeting?
- What sequencer did you use?
- What kind of extraction did you perform? (Kit manual etc)
- Do you have nanodrop/qubit 
