---
title: 'Are Primers Present in Ade'
date: 2023-05-25T13:29:43Z
draft: false
meta_img: "images/image.png"
tags:
  - "Ade"
  - "visualize-ampliseq"
  - "Mock"
  - "Module"
  - "primer-detect"
description: "Description for the page"
---

### Todos for Today

- Jackwood Blast
  - meet Ben and Brian TBD
  - try out a local blast search to see memory, cpu time limitation
- Ade
  - Try to detect the primers present
  - Use cutadapt to remove primers
  - How does the other Ben's Analysis line up with mine/ampliseq?
    - filtering step for the abundance?
    - Filtering unknown taxa?
  - Run a proper analysis to send to Ade
- gg-catalog
  - Generate a gene network 
    - how do you do this?
      - possibly this: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-3371-7
  - query the KO list and cross ref to the abundance data
  - Find a Shotgun Analysis involved with chickens and some kind of phenotypic data.

### Ade

#### Picking proper parameters

Who was filtered out of the negative control analysis?



What was the accuracy of the mock analysis?

[] Re-run the analysis with HQ and New Visualize ampliseq?
[] figure out what primers are present?
  - https://github.com/linsalrob/primer-trimming
  - https://github.com/OpenGene/fastp
  - 
[] re-run with cutadapt

#### Primer-trimming


#### Fastp



#### Ben's Other analysis

What steps did he take?

What steps do I take?



#### Longitudinal Analysis

See if it makes a difference in this specific analysis. 


