---
title: "Project Check in and Examining Ampliseq Results"
author: "Ben Lorentz"
date: '2022-11-02'
slug: project-check-in-and-examining-ampliseq-results
categories: []
tags:
- one tag
- another tag
meta_img: images/image.png
description: Description for the page
---

### Todos for Today:
- continue reading jones
- Visualize Ampliseq
  - Refine chunk 06
  - add example params and slurm for ampliseq into ampliseq-vis repos
- nf-core/Ampliseq
  - compare low, medium, high richness results
- re-watch the lecture for ChIP-seq
- What is a core microbiome?
  - Can we use this to find the major players in chicken gut segments?
  - are keystone species more important in gut microbiomes?
- How can multi-omics projects be implemented in host-microbe interactions
- Check in on classifier still running

---

- Try to run DESeq2's GLM/LRT on shaile's design
- Learn a little more about GLM in R

---

- BIOSQL, SQL or Mongo DB tutorials
- Genome Assembly from Isolates
- Kelly Shotgun Metagenomic Data
- good example 16s data to hone parameters


I think we should spend like 2 sections of a pom on general GLM and then if we aren't getting far then we can pivot to examining the results from ampliseq, and report 06/ the rest in that batch. Then we can work on outlining the project description for my term paper. 

### Notes on Broliers

"Conventional broiler (meat) chickens reach a final weight of 6.6 lbs (3kg) in approximately ***40-44 days***. Across their lifetime, these fast-growing birds gain an average of 68-75 g per day, with daily gain increasing with age (Aviagen, 2022; Cobb, 2022)." - https://www.poultry-welfare-extension.com/uploads/2/5/6/3/25631086/pec_newsletter_nov22__1_.pdf

### General GLM in R

"GLM in R is a class of regression models that supports non-normal distributions and can be implemented in R through glm() function that takes various parameters, and allowing user to apply various regression models like logistic, poission etc., and that the model works well with a variable which depicts a non-constant variance, with three important components viz. random, systematic, and link component making the GLM model, and R programming allowing seamless flexibility to the user in the implementation of the concept." - https://www.educba.com/glm-in-r/

I think that [this thread](https://www.seqanswers.com/forum/bioinformatics/bioinformatics-aa/42269-deseq2-model-matrix-formula) might explain how to compare difference in groups in tissue compared to general difference in groups. 

I went and got three books:
  - Linear Mixed Models West Welch and Galecki 2007
  - Using Multivariate Statistics Tabachnick and Fidell 1996
  - Optimum Design for Multi-Factor Models Schwabe 1996
  
We'll see if there's anything really helpful in them.

Okay: I think that this is the way I'm gonna approach the analysis:

"Interactions

Interaction terms can be added to the design formula, in order to test, for example, if the log2 fold change attributable to a given condition is different based on another factor, for example if the condition effect differs across genotype.

Initial note: Many users begin to add interaction terms to the design formula, when in fact a much simpler approach would give all the results tables that are desired. We will explain this approach first, because it is much simpler to perform. If the comparisons of interest are, for example, the effect of a condition for different sets of samples, a simpler approach than adding interaction terms explicitly to the design formula is to perform the following steps:

    combine the factors of interest into a single factor with all combinations of the original factors
    change the design to include just this factor, e.g. ~ group

Using this design is similar to adding an interaction term, in that it models multiple condition effects which can be easily extracted with results. Suppose we have two factors genotype (with values I, II, and III) and condition (with values A and B), and we want to extract the condition effect specifically for each genotype. We could use the following approach to obtain, e.g. the condition effect for genotype I:" - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#contrasts


### Ben Jackwood

Ben stopped by to ask about trying to align a Gene HAT2 or GCS1 to an Eimeria Maxima reference genome, but the genome was only in contigs so when he tried to just aligh the sequence to the whole genome it took forever/wouldn't finsh. I suggested he blast his query sequence against the reference genome and he recieved 2 high-quality hits. He pulled those regions out and used MAFFT and Clustalw to align the query sequence to the reference and was able to get some okay alignments but the repetitive elements make it a little difficult to line up well. I suggested he could also try BWA-aligner if he wanted to get into cli tools. 


### Examine nf-core/Ampliseq

Low Richness successfully completed and was downloaded to the ssd.

```{r message=FALSE}

low<-read.csv("/mnt/content/tables_for_ampliseq/low_richness_stats.tsv",sep = '\t')
order_low <- low[order(low$filtered_tax_filter,decreasing = T),]
knitr::kable(order_low)

```

Medium Richness failed halfway through

```bash
Error executing process > 'NFCORE_AMPLISEQ:AMPLISEQ:QIIME2_INASV (ASV_table.tsv)'

Caused by:
  java.nio.file.FileAlreadyExistsException: /scratch/bjl34716/nf_dev/ampliseq-benchmark/work/singularity/quay.io-qiime2-core-2021.8.img
```

```{r}


#read_table("/mnt/content/tables_for_ampliseq/med_richness_stats.tsv")

```

re-started
slurm: 15165472
nextflow: cranky_kalam

High Richness

```{r message=FALSE}

high<-read.csv("/mnt/content/tables_for_ampliseq/high_richness_stats.tsv",sep='\t')
order_high <- high[order(high$filtered_tax_filter,decreasing = T),]
knitr::kable(order_high)

```


