---
title: Background Reading for Gut Microbiome Interaction
author: Ben Lorentz
date: '2022-11-07'
slug: background-reading-for-gut-microbiome-interaction
categories: []
tags:
  - one tag
  - another tag
meta_img: images/image.png
description: Description for the page
---

### Todos for Today:

- continue reading Tillocca
- continue reading jones
- Visualize Ampliseq
  - add example params and slurm for ampliseq into ampliseq-vis repos
- Run Visualize Ampliseq
  - on low med high richness samples
- re-watch the lecture for ChIP-seq
- What is a core microbiome?
  - Can we use this to find the major players in chicken gut segments?
  - are keystone species more important in gut microbiomes?
- How can multi-omics projects be implemented in host-microbe interactions
- Check in on classifier still running

---

- Try to run DESeq2’s GLM/LRT on shaile’s design
- Learn a little more about GLM in R

---

- BIOSQL, SQL or Mongo DB tutorials
- Genome Assembly from Isolates
- Kelly Shotgun Metagenomic Data
- good example 16s data to hone parameters

The temptation for today is to dig really deep into visualizing the 3 richness depths, but I want to put that off for just a little bit while I try to read some more papers today to start to get ahead on that to have some more information to report to Aggrey soon. A third task that would also be good to dig into would be running one of my tissues for the term paper. 

I was able to finish reading Tillocca, I think it gave some really good background ideas on what kind of taxa are generally present. 

Todos from this:
  - look at references and see if we can get a generalized taxa table
  - from the taxa table functionally annotate the 'role' of the taxa present in each segement
    - they may have done some of this heavy lifting for us.
    
Shailes sent us his data to analyse so that can be a task for this week.

I set up a [repo/site](https://lorentz-host-microbe-interaction.netlify.app/) for this microbial interaction stuff! I want to find a better homepage format than just chronological pages, but that's the next step for them.

### Todos for Tomorrow:

- Tillocca
  - look at references and see if we can get a generalized taxa table
  - from the taxa table functionally annotate the 'role' of the taxa present in each segement
    - they may have done some of this heavy lifting for us.
- site for microbiome
  - find better index page
  - continue to migrate data from synology dir to the ssd one and push
- continue reading jones
- Visualize Ampliseq
  - add example params and slurm for ampliseq into ampliseq-vis repos
- Run Visualize Ampliseq
  - on low med high richness samples
- re-watch the lecture for ChIP-seq
- What is a core microbiome?
  - Can we use this to find the major players in chicken gut segments?
  - are keystone species more important in gut microbiomes?
- How can multi-omics projects be implemented in host-microbe interactions
- Check in on classifier still running

---

- Try to run DESeq2’s GLM/LRT on shaile’s design

---

- BIOSQL, SQL or Mongo DB tutorials
- Genome Assembly from Isolates
- Kelly Shotgun Metagenomic Data
- good example 16s data to hone parameters